backend/src/gtfs/gmd.rs

```diff
@@ -1,74 +1,80 @@
-use std::collections::BTreeMap;
-use std::fs::File;
-use std::io::BufWriter;
-
-use anyhow::Result;
+use anyhow::{bail, Result};
 use chrono::NaiveTime;
-use flatgeobuf::{
-    ColumnType, FeatureProperties, FgbCrs, FgbFeature, FgbWriter, FgbWriterOptions, GeometryType,
-    GeozeroGeometry, HttpFgbReader,
-};
-use geo::{Contains, LineString};
-use geozero::{ColumnValue, PropertyProcessor};
+use futures_util::StreamExt;
+use geo::{Contains, Coord, LineString};
+use geomedea::{Bounds, Geometry, LngLat, PropertyValue};
 use serde::{Deserialize, Serialize};
 use utils::Mercator;
 
 use super::{orig_ids, GtfsModel, Route, RouteID, Stop, StopID, Trip};
 use crate::graph::RoadID;
 
 impl GtfsModel {
-    pub fn to_fgb(&self, filename: &str) -> Result<()> {
-        let mut fgb = FgbWriter::create_with_options(
-            "gtfs",
-            GeometryType::LineString,
-            FgbWriterOptions {
-                crs: FgbCrs {
-                    code: 4326,
-                    ..Default::default()
-                },
-                ..Default::default()
-            },
-        )?;
-        // TODO JSON?
-        fgb.add_column("data", ColumnType::Binary, |_, col| {
-            col.nullable = false;
-        });
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn to_geomedea(&self, filename: &str) -> Result<()> {
+        use std::fs::File;
+        use std::io::BufWriter;
+
+        use geomedea::{Feature, Geometry, Properties, PropertyValue, Writer};
+
+        let out = BufWriter::new(File::create(filename)?);
+        let mut writer = Writer::new(out, true)?;
 
         for (variant, linestring) in group_variants(self) {
-            fgb.add_feature_geom(geo::Geometry::LineString(linestring), |feature| {
-                // TODO bincode or something else?
-                let json = serde_json::to_vec(&variant).unwrap();
-                // TODO Play with compression levels
-                let compressed = miniz_oxide::deflate::compress_to_vec(&json, 10);
-
-                feature
-                    .property(0, "data", &ColumnValue::Binary(&compressed))
-                    .unwrap();
-            })?;
+            // TODO geozero to this translation better?
+            let geom = Geometry::LineString(geomedea::LineString::new(
+                linestring
+                    .0
+                    .into_iter()
+                    .map(|pt| geomedea::LngLat::degrees(pt.x, pt.y))
+                    .collect(),
+            ));
+            let mut props = Properties::empty();
+            // TODO bincode or something else?
```

Comment:
The size difference is probably coming from here, I need to rethink how to encode this. Most of the size comes from a bunch of `chrono::NaiveTime`s right now, which get encoded in a pretty naive way

Reply:
Is `variant` some pre-encoding of all your properties into a single byte stream?

Internally geomedea is using bincode for encoding property and geometry, and then each page is zstd compressed.

Varints probably make sense for property data, but probably not currently for geometry data, until/unless I also implement delta encoding for geometries. I'd like to implement delta encoding but it'll require reworking some API internals.

Reply:
I'm using `serde_json::to_vec` on ```
struct RouteVariant {
    // Per stop, (original ID and name)
    pub stop_info: Vec<(orig_ids::StopID, String)>,

    // Each one has an arrival time per stop
    pub trips: Vec<Vec<NaiveTime>>,

    // Metadata
    pub route: Route,
}
```
The space is dominated by the times. Dropping some precision and some delta encoding would make tons of sense there.

I'll also play with using `PropertyValue::Vec` of some integers manually, instead of this.

Reply:
https://docs.rs/chrono/latest/src/chrono/naive/time/serde.rs.html#5

OK, NaiveTime gets serde-ified as a string right now, that's amazing. Switching to something more appropriate...